Providers/Profile.swift

```diff
@@ -782,8 +806,59 @@ public class BrowserProfile: Profile {
             }
         }
 
+        @objc func onStartSyncing(notification: NSNotification) {
+            syncDisplayState = .InProgress
+        }
+
         @objc func onFinishSyncing(notification: NSNotification) {
-            self.lastSyncFinishTime = NSDate.now()
+            if let syncState = syncDisplayState where syncState == .Good {
+                self.lastSyncFinishTime = NSDate.now()
+            }
+        }
+
+        private func displayStateForSyncResult(result: SyncResult) -> SyncDisplayState {
+            guard result.value.isSuccess,
+            let syncState = result.value.successValue else {
+                return SyncDisplayState.Bad(message: Strings.FirefoxSyncFailedTitle)
+            }
+
+            return displayStateForSyncState(syncState)
+        }
+
+        private func displayStateForEngineResults(result: Maybe<EngineResults>?) -> SyncDisplayState {
+            guard let result = result else {
+                return .Good
+            }
+            guard result.isSuccess,
+                let results = result.successValue else {
+                    return SyncDisplayState.Bad(message: Strings.FirefoxSyncFailedTitle)
+            }
+            let errorResults: [SyncDisplayState]? = results.flatMap { identifier, status in
+                let displayState = self.displayStateForSyncState(status, identifier: identifier)
+                return displayState == .Good ? nil : displayState
+            }
+            return (errorResults?.isEmpty ?? true) ? .Good : (errorResults?.first)!
```

Comment:
I wonder if this is the right place to coalesce the engine errors into a single sync state. I remember this morning I was advocating for simplifying the reporting of the error into a single 'sync failed' state but I wonder if this is where we should do it. What do you think about keeping a mapping of engine -> sync state and we add a computed property to infer the overall sync status so we don't lose the status of each engine? I'm not sure.